### PR TITLE
Fix to Bug #65 - Added top recent links API

### PR DESCRIPTION
**Pull request by mzhilyaev**
Tests are passing OK.
Couldn't find any relevant example to test getting places data from the content side.
If such test is needed, please advice where to look

### CODE REVIEW

File: content-src/actions/action-manager.js [hunk 1]
```diff
@@ -4,6 +4,8 @@ const ActionManager = require("../lib/ActionManager");
 const am = new ActionManager([
   "TOP_FRECENT_SITES_REQUEST",
   "TOP_FRECENT_SITES_RESPONSE",
+  "TOP_RECENT_LINKS_REQUEST",
```
Comment (oyiptong) [inline content-src/actions/action-manager.js]: I think we can drop the 'TOP' from the name

File: lib/ActivityStreams.js [hunk 2]
```diff
@@ -52,9 +52,9 @@ ActivityStreams.prototype = {
   },
 
   /**
-   * Responds to places requests
+   * Responds to places frecent sites requests
    */
-  _respondToPlacesRequests(msgName, params) {
+  _respondToFrecentSitesRequests(msgName, params) {
```
Comment (oyiptong) [inline lib/ActivityStreams.js]: Please move this to a switch case statement, like in #110

File: lib/PlacesProvider.js [hunk 3]
```diff
@@ -134,6 +134,53 @@ Links.prototype = {
...
+    if (!options) {
+      options = {};
+    }
+
+    let {limit} = options;
+    if (!limit || limit.options > HISTORY_RESULTS_LIMIT) {
+      limit = HISTORY_RESULTS_LIMIT;
+    }
+
+    let sqlQuery = "SELECT moz_places.url, moz_favicons.data as favicon, mime_type, title, " +
```
Comment (oyiptong) [inline lib/PlacesProvider.js]: Can you please use back ticks? They are the new multi line strings

File: lib/PlacesProvider.js [hunk 4]
```diff
@@ -134,6 +134,53 @@ Links.prototype = {
...
+                   "      frecency, last_visit_date as lastVisitDate,  \"history\" as type " +
+                   "FROM moz_places " +
+                   "LEFT JOIN moz_favicons " +
+                   "ON favicon_id = moz_favicons.id " +
+                   "WHERE hidden = 0 AND last_visit_date NOTNULL " +
+                   "ORDER BY lastVisitDate DESC, frecency DESC, url " +
+                   "LIMIT :limit";
+
+    let links = yield this.executePlacesQuery(sqlQuery, {
+                  columns: ["url", "title", "lastVisitDate", "frecency"],
```
Comment (oyiptong) [inline lib/PlacesProvider.js]: You need the Favicon columns

File: lib/PlacesProvider.js [hunk 5]
```diff
@@ -134,6 +134,53 @@ Links.prototype = {
...
+                  params: {limit: limit}
+                });
+
+    // @TODO - this the exact same code as getTopFrecentSites uses
```
Comment (oyiptong) [inline lib/PlacesProvider.js]: Do not leave this as a TODO. In fact, the function is extracted in #110. Rebase on top when it is merged?

File: lib/PlacesProvider.js [hunk 6]
```diff
@@ -134,6 +134,53 @@ Links.prototype = {
...
 
   /**
+   * Gets the top recent links
+   *
+   * @param {Integer} options
+   *                  limit: Maximum number of results to return. Max 100.
+   *
+   * @returns {Promise} Returns a promise with the array of links as payload.
+   */
+  getTopRecentLinks: Task.async(function*(options) {
```
Comment (oyiptong) [inline lib/PlacesProvider.js]: GetRecentLinks should be fine I think

File: test/test-PlacesProvider.js [hunk 7]
```diff
@@ -94,6 +94,52 @@ exports.test_Links_getTopFrecentSites_Order = function*(assert) {
   }
 };
 
+exports.test_Links_getTopRecentLinks = function*(assert) {
```
Comment (oyiptong) [inline test/test-PlacesProvider.js]: GetRecentLinks

**Comment by mzhilyaev** [comment]
Cancelling this pull request and re-openning another one re-based on top of 30_bookmarks_api, this should all just work

**Pull request closed without merging**